src/main/java/hudson/util/VersionNumber.java

```diff
@@ -513,16 +513,25 @@ public boolean isNewerThan(VersionNumber rhs) {
         return compareTo(rhs) > 0;
     }
 
+    /**
+     * Returns a digit (numeric component) by its position.
+     *
+     * @param idx Digit position we want to retrieve starting by 1 (not zero).
+     * @return The digit or -1 in case the position does not correspond with a digit.
+     */
     public int digit(int idx) {
-        Iterator i = items.iterator();
-        Item item = (Item) i.next();
-        while (idx > 0 && i.hasNext()) {
+        Iterator it = items.iterator();
+        int i = 1;
+        Item item = null;
+        while (i <= idx && it.hasNext()) {
+            item  = (Item) it.next();
             if (item instanceof IntegerItem) {
-                idx--;
+                i++;
+            } else {
+                return -1;
```

Comment:
Is it possible getting a non-integer element with later elements being actually integers? Because if it is, this would not be contemplating them.